### PR TITLE
[wip] API v2

### PR DESCRIPTION
**Pull request by callum-oakley**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -23,10 +23,10 @@ export const appendQueryParams = (queryParams, url) => {
   return url + separator + urlEncode(queryParams)
 }
 
-export const appendQueryStringArrayParams = (key, values, url) => {
-  const separator = contains('?', url) ? '&' : '?'
+export const appendQueryParamsAsArray  = (key, values, url) => {
+  const separator = contains('?', url) ? '' : '?'
```
Comment (hamchapman) [inline src/utils.js]: Why did this line change? Don't we still want to add a `&` if there is already a `?` present in the URL?

**Comment by callum-oakley** [comment]
I think this is pretty much there now. There is one slighly flaky test, the cause of which is eluding me... (See the FIXME in tests/main.js)

File: src/user-store.js [hunk 2]
```diff
@@ -29,7 +30,13 @@ export class UserStore {
     this.store.initialize(map(this.decorate, initial))
   }
 
-  set = (userId, basicUser) => this.store.set(userId, this.decorate(basicUser))
+  set = (userId, basicUser) => {
+    return this.store.set(
+      userId,
+      this.decorate(basicUser)
+    )
+      .then(() => forEach(hook => hook(userId), this.onSetHooks))
```
[the remark this answers is not in the document]
Comment (lukejacksonn) [reply src/user-store.js]: Ohh duh. Yes that makes sense.

File: example/main.js [hunk 3]
```diff
@@ -12,12 +12,36 @@ const tokenProvider = new Chatkit.TokenProvider({
...
+  onUserLeftRoom: (room, user) => {
+    console.log('user: ', user, ' left room: ', room)
+  },
+  onPresenceChanged: ({ previous, current }, user) => {
+    console.log('user: ', user, ' was ', previous, ' but is now ', current)
+  }
+})
   .then(cUser => {
     currentUser = cUser
+    window.currentUser = cUser
```
[the remark this answers is not in the document]
Comment (hamchapman) [reply example/main.js]: Just for testing really

**Pull request merged**